bigtop-deploy/puppet/modules/ranger/manifests/init.pp

```diff
@@ -17,57 +17,77 @@
 
   class deploy($roles) {
     if ('ranger-server' in $roles) {
-      include ranger::prerequisites
       include ranger::admin
     }
   }
 
-  class prerequisites {
+
+  class admin($admin_password) {
     # Before Facter 3.14.17, Rocky Linux 8 is detected as 'RedHat'.
     # https://puppet.com/docs/pe/2019.8/osp/release_notes_facter.html#enhancements-3-14-17
-    if (($operatingsystem == 'RedHat' or $operatingsystem == 'Rocky') and 0 <= versioncmp($operatingsystemmajrelease, '8')) {
-      # For some reason, 'python3' doesn't seem to work on Rocky Linux 8.
-      $python = 'python36'
-    } else {
-      $python = 'python3'
+    case $operatingsystem {
+      /(?i:(ubuntu|debian))/: {
+            $postgres_packages = ['postgresql']
+            $python = 'python3'
+            $java_home_env = 'JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk-amd64'
+        }
+        default: {
+            $postgres_packages = ['postgresql-jdbc', 'postgresql-server']
+            $python = 'python36'
```

Comment:
After applying this PR and #1201, running smoke test on CentOS 7 fails as follows. Looks like `python36` doesn't work on CentOS 7 and we have to use `python3` as a package name. Would you fix it?

```
$ ./docker-hadoop.sh -d -dcp -C config_centos-7.yaml -G -L -k hdfs,hbase,ranger -s ranger -c 1

...

Error: Could not update: Execution of '/usr/bin/yum -d 0 -e 0 -y list python36' returned 1: Error: No matching Packages to list                                                               
Wrapped exception:                                                                                                                                                                            
Execution of '/usr/bin/yum -d 0 -e 0 -y list python36' returned 1: Error: No matching Packages to list                                                                                        
Error: /Stage[main]/Ranger::Admin/Package[python36]/ensure: change from absent to latest failed: Could not update: Execution of '/usr/bin/yum -d 0 -e 0 -y list python36' returned 1: Error: N
o matching Packages to list                                                                                                                                                                   
Notice: /Stage[main]/Ranger::Admin/Package[ranger-admin]/ensure: created                                                                                                                      
Notice: /Stage[main]/Ranger::Admin/Exec[initdb]: Dependency Package[python36] has failures: true                                                                                              
Warning: /Stage[main]/Ranger::Admin/Exec[initdb]: Skipping because of failed dependencies
Notice: /Stage[main]/Ranger::Admin/Service[postgresql]: Dependency Package[python36] has failures: true
Warning: /Stage[main]/Ranger::Admin/Service[postgresql]: Skipping because of failed dependencies
Notice: /Stage[main]/Ranger::Admin/Exec[change_postgres_password]: Dependency Package[python36] has failures: true
Warning: /Stage[main]/Ranger::Admin/Exec[change_postgres_password]: Skipping because of failed dependencies
Notice: /Stage[main]/Ranger::Admin/File[/usr/lib/ranger-admin/install.properties]: Dependency Package[python36] has failures: true
Warning: /Stage[main]/Ranger::Admin/File[/usr/lib/ranger-admin/install.properties]: Skipping because of failed dependencies
Notice: /Stage[main]/Ranger::Admin/Exec[/usr/lib/ranger-admin/setup.sh]: Dependency Package[python36] has failures: true
Warning: /Stage[main]/Ranger::Admin/Exec[/usr/lib/ranger-admin/setup.sh]: Skipping because of failed dependencies
Notice: /Stage[main]/Ranger::Admin/Exec[/usr/lib/ranger-admin/set_globals.sh]: Dependency Package[python36] has failures: true
Warning: /Stage[main]/Ranger::Admin/Exec[/usr/lib/ranger-admin/set_globals.sh]: Skipping because of failed dependencies
Notice: /Stage[main]/Ranger::Admin/Exec[systemctl daemon-reload]: Dependency Package[python36] has failures: true
Warning: /Stage[main]/Ranger::Admin/Exec[systemctl daemon-reload]: Skipping because of failed dependencies
Notice: /Stage[main]/Ranger::Admin/Service[ranger-admin]: Dependency Package[python36] has failures: true
Warning: /Stage[main]/Ranger::Admin/Service[ranger-admin]: Skipping because of failed dependencies
```